### ISSUE TITLE
[Firefox] uBlock does not block popup anymore under a rare circumstance

### ISSUE THREAD

**Issue by g3939434** [gorhill/uBlock#450]
uBlock 0.9.9.3, Firefox 39.0, Windows 8.1

Steps to Reproduce:
1. Open logger and type in `|popup` in filter.
2. Go to [Google's News](https://news.google.com/nwshp) and click any news' link.
3. Logger records popup successfully.
4. Close the news' link you clicked.
5. Open another [Google's News](https://news.google.com/nwshp) in a new tab.
6. Click any news' link in the **new** tab and then close the news' link you clicked.
7. Close the **old** Google's News tab.
8. Repeat Step 5 to 7 one time.
9. Click any news' link.
10. Logger does not record any popup.

If you cannot reproduce, you may repeat Step 5 to 7, then Step 9 & 10. It may be needed to repeat a few times to reproduce.

I can only reproduce in Firefox, not in Chrome.

**Edit**:

After further testing, the fact is uBlock does not block any popup completely. It fails all the [popup tests](http://raymondhill.net/ublock/popup.html) - tested with rule `no-popups: raymondhill.net true`

So I believe it is not a problem of logger but the core engine.

**Comment by Betsy25** [comment]
Cannot reproduce the problem, but regarding the popups, It failed *all* ![popup tests](http://raymondhill.net/ublock/popup.html) here also.

Firefox nightly v42, uBlock 0.9.9.4, Win7-64bit

**Comment by gorhill** [comment]
> It fails all the popup tests - tested with rule `no-popups: raymondhill.net true`

The popups were blocked correctly on my side (with or without `no-popups: raymondhill.net true`), FF 38 or Nightly 41.

**Comment by CrisBRM** [comment]
@gorhill Tested on W8.1, everything just worked™.

**Comment by gorhill** [comment]
Yes, I just tested with stock 0.9.9.3, FF39 Win7 and all popups were blocked.

**Comment by g3939434** [comment]
@gorhill Did you just test the popup tests directly or after the steps?

Since uBlock can block all popups at first, but not after the steps I stated.

**Comment by gorhill** [comment]
> Since uBlock can block all popups at first, but not after the steps I stated

Ok. I went with @Betsy25's comment, which suggested that popups are no longer blocked regardless.

**Comment by g3939434** [comment]
Eventually, do you reproduce anything?

**Comment by gorhill** [comment]
Yes I did reproduce the absence of `popup` entries in the logger, after repeating 5-7 many times.

**Comment by Betsy25** [comment]
Proplem seems to be "fixed", following the procedure of the OP, and then testing out the popup tests,
only tests 6,8, 11 'don't work' as intended.

**Comment by g3939434** [comment]
I confirmed the problem was fixed in v0.9.9.4-dev.5.

The problem was caused by closing tabs shortly. Therefore, a clearer step 6 should be - "... then close the news' link you clicked **immediately**".

@Betsy25: You have an old profile with an old whitelist. You should remove `data-scheme` and `file-scheme` from whitelist and temporarily remove `about-scheme` when testing Test 7 with Chrome. (Note: Firefox blocked it even with not whitelisting `about-scheme`) - reference [chrisaljoudi #510](https://github.com/chrisaljoudi/uBlock/issues/510), [#663](https://github.com/chrisaljoudi/uBlock/issues/663)

A default profile should look like this:
```
behind-the-scene
about-scheme
chrome-extension-scheme
chrome-scheme
loopconversation.about-scheme
opera-scheme
```

Besides the controversy of Test 8, which Chrome classifies it as a popup and closes it when clicking.
Firefox classifies it as a doc and remains open. A strange behavior occurred with Chrome when testing Test 11. Without removing `data-scheme` from whitelist, I expected Chrome should fail the test with result "not ok". However, Chrome passed the test with result "ok" but logger showed that xhr was not blocked.
